src/cryptography/hazmat/backends/openssl/hashes.py

```diff
@@ -22,8 +22,8 @@ def __init__(self, backend, algorithm, ctx=None):
             ctx = self._backend._ffi.gc(
                 ctx, self._backend._lib.Cryptography_EVP_MD_CTX_free
             )
-            evp_md = self._backend._lib.EVP_get_digestbyname(
-                algorithm.name.encode("ascii"))
+            name = self._backend._build_openssl_digest_name(algorithm)
+            evp_md = self._backend._lib.EVP_get_digestbyname(name)
             if evp_md == self._backend._ffi.NULL:
                 raise UnsupportedAlgorithm(
                     "{0} is not a supported hash on this backend.".format(
```

Comment:
This is going to be a bad error message if you try to use an alternate digest size